### PR TITLE
[OFFAPPS-784] Fixed counting bugs

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description
Fixed bugs found bye alex QA.

### References
* JIRA: https://zendesk.atlassian.net/browse/OFFAPPS-784

### Risks
* [low] pausing still broken in some configurations.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -546,14 +551,23 @@
 
     resume: function() {
       if (!this.isPaused()) return;
```
Comment (nick-f) [inline app.js]: This is also the first line in calcPausedTime(). Could it be removed here or is it in both places for clarity?

Comment (ocke) [inline app.js]: Explained in 🍗

File: app.js [hunk 2]
```diff
@@ -533,6 +533,11 @@
       this.startTime = getTick();
       this.elapsedPausedTime = 0;
       this.pausedAt = 0;
+
+      if (this.wasPaused) {
```
Comment (sandlerr) [inline app.js]: doesn't this belong in `commitTicketTime`? Doesn't really fit the name of this function

File: app.js [hunk 3]
```diff
@@ -546,14 +551,23 @@
...
+      this.calcPausedTime();
       this.pausedAt = 0;
     },
 
+    // Sets the amount of time paused.
+    // If called again, make sure we don't calculate the paused time double.
+    calcPausedTime: function() {
+      if (!this.isPaused()) return;
+      var tick = getTick();
+      this.elapsedPausedTime += tick - this.pausedAt;
```
Comment (sandlerr) [inline app.js]: I'd kinda prefer if we only ever modify this value when you hit resume. Is it possible to just return a count from this function instead?

File: app.js (hunk 2 above)
Comment (ocke) [inline app.js]: yes, that does make more sense :+1:

File: app.js [hunk 4]
```diff
@@ -546,30 +548,46 @@
...
-      this.totalTime(this.totalTime() + ticketTime);
+      // only update if ticketTime is > 0
+      if (ticketTime) {
+        this.ticketFieldTime(ticketTime);
+        this.ticketFieldTotalTime(this.ticketFieldTotalTime() + ticketTime);
+      }
 
       this.resetNewTimers();
+
+      if (wasPaused) {
```
Comment (sandlerr) [inline app.js]: maybe add a comment here why we have to do this?

Comment (ocke) [inline app.js]: ok

Comment (ocke) [inline app.js]: I put the comment where i create the var

File: app.js [hunk 5]
```diff
@@ -546,30 +548,47 @@
...
 
-      return Math.floor((ticketTime - this.elapsedPausedTime) / 1000);
+      return Math.floor((ticketTime - pausedTime) / 1000);
     },
 
     commitTicketTime: function(ticketTime) {
-      ticketTime = ticketTime || this.ticketTime();
+      ticketTime = ticketTime !== undefined ? ticketTime : this.elapsedTimeV2();
+
+      // For stay on ticket, we don't want the timer to start counting again.
```
Comment (sandlerr) [inline app.js]: `... and calling `resetNewTimers` resets the pause state`

**Pull request merged**